android/source/src/main/kotlin/com/gu/source/components/banner/AlertBanner.kt

```diff
@@ -0,0 +1,315 @@
+            horizontalArrangement = Arrangement.spacedBy(AlertBanner.Style.iconTextSpacing),
+        ) {
+            priority.getIcon()?.let {
+                Icon(
+                    imageVector = it,
+                    tint = priority.getIconTint().current,
+                    contentDescription = null,
+                )
+            }
+
+            Text(
+                text = text,
+                style = AlertBanner.Style.textStyle,
+                color = priority.getTextColor().current,
+            )
+        }
+        Icon(
+            imageVector = AlertBanner.Style.iconCross,
+            tint = priority.getCrossIconTint().current,
+            contentDescription = null,
+            modifier = Modifier
+                .clickable(onClick = onCloseClick),
+        )
+    }
+}
+
+/**
+ * A composable function that displays an alert banner with optional clickable links.
+ *
+ * @param annotatedText The text to be displayed in the banner having link in it.
+ * @param priority The type of message to be displayed (Neutral, Informative, Error).
+ * @param onCloseClick A callback to be invoked when the cancel icon is clicked.
+ * @param onClick A callback to be invoked when the banner is clicked.
+ * @param modifier The modifier to be applied to the banner.
+ */
+@SuppressLint("DiscouragedApi")
+@Composable
+fun AlertBanner(
+    annotatedText: AnnotatedString,
+    priority: AlertBanner.Priority,
+    onCloseClick: () -> Unit,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Row(
+        modifier = modifier
+            .fillMaxWidth()
+            .background(priority.getBackdropColour().current)
+            .padding(AlertBanner.Style.ContentPadding),
+        horizontalArrangement = Arrangement.spacedBy(AlertBanner.Style.crossTextSpacing),
+    ) {
+        Row(
+            modifier = Modifier
+                .weight(1f)
+                .clickable { onClick() },
+            horizontalArrangement = Arrangement.spacedBy(AlertBanner.Style.iconTextSpacing),
+        ) {
+            priority.getIcon()?.let {
+                Icon(
+                    imageVector = it,
+                    tint = priority.getIconTint().current,
+                    contentDescription = null,
+                )
+            }
+
+            Text(
+                text = annotatedText,
+                style = AlertBanner.Style.textStyle,
+                color = priority.getTextColor().current,
+            )
+        }
+        Icon(
+            imageVector = AlertBanner.Style.iconCross,
+            tint = priority.getCrossIconTint().current,
+            contentDescription = null,
+            modifier = Modifier
+                .clickable(onClick = onCloseClick),
+        )
+    }
+}
+
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+@SuppressLint("DiscouragedApi")
+@PreviewPhoneBothMode
+@Composable
+internal fun AlertBannerWithTextPreview() {
+    AppColourMode {
+        SourceCoreTheme {
```

Comment:
You don't need this here because you aren't reading any theme specific values.

Reply:
done